frontend/packages/yki/src/redux/selectors/examSessions.ts

```diff
@@ -37,40 +37,19 @@ const filterExamSessions = (
     );
   }
 
-  if (filters.excludeFullSessions) {
-    filteredData = filteredData.filter(
-      (es) =>
-        es.participants < es.max_participants ||
-        (es.post_admission_active &&
-          es.post_admission_quota &&
-          es.pa_participants < es.post_admission_quota)
-    );
-  }
-
-  if (filters.excludeNonOpenSessions) {
-    const now = dayjs();
-    filteredData = filteredData.filter(
-      (es) => isRegistrationOpen(es, now) || isPostAdmissionOpen(es, now)
-    );
+  if (filters.excludeFullSessions || filters.excludeNonOpenSessions) {
+    filteredData = filteredData.filter((es) => {
+      const { open, availablePlaces } =
+        ExamSessionUtils.getCurrentOrFutureAdmissionPeriod(es);
+      if (filters.excludeFullSessions && !availablePlaces) {
```

Comment:
Vois heittää ehkä rivinvaihdon ennen if-elsejä

Reply:
Ja käyttäjällehän sellaiset sessiot, joiden rekisteröintiaika mennyt umpeen näkyvät kai täysinä? Tai onko tuota muutettu uudessa käyttöliittymässä? Mietin että pitäiskö tässä filteröidä pois myös tuollaiset sessiot, vaikka niissä olisikin paikkoja vapaina.

Edit:
Piti tarkistaa niin tosiaan joo korjattu tänne uuteen ui:hin tämä. Jätetään siis tietysti näin.

Ja tän perusteella kun katsoinkin just tuota toteuttamaani `compareExamSessions` logiikkaa, niin siellä siis ollut ajatuksena että `compareExamSessionsByRoom` käsittelee täysinä sellaisia sessioita, joiden ilmoittautumisaika mennyt umpeen. Ei tarve tän pullarin sisällä välttis yhtenäistää mut vois olla ehkä hyvä "korjata" tuota koodilogiikkaa että `hasRoom` tosiaan palauttaisi `true` jos tilaisuuteen on paikkoja vaikka ilmoaika olisikin mennyt umpeen.